.github/workflows/spark-ci.yml

```diff
@@ -69,9 +69,15 @@ jobs:
     runs-on: ubuntu-22.04
     strategy:
       matrix:
-        jvm: [8, 11, 17]
+        jvm: [8, 11, 17, 21]
         spark: ['3.3', '3.4', '3.5']
         scala: ['2.12', '2.13']
+        exclude:
+          # Spark 3.5 is the first version supporting Java 21 (https://issues.apache.org/jira/browse/SPARK-42369)
```

Comment:
Actually Spark 3.5 does not fully support Java 21, the right ticket is https://issues.apache.org/jira/browse/SPARK-43831

Reply:
Thanks @pan3793 for your comment and the link. This is good reference.
If our tests with Spark 3.5 pass with Java 21, we don't need to disable them, but it's good to know we should be expecting challenges. I will rework the comment.